woocommerce/woocommerce
# Adding more menu with items using slot fill

### All Submissions:






### Changes proposed in this Pull Request:

Adding the standard more menu to the block product editor header. This is done by:
- Adding two slot fills, one for header items and another for the more menu items.
- Refactoring the more menu items in the `/client/product` folder.
- Using slot fill to bring those into the block editor while preserving function on the old-new editor. 

Closes #37097  .

### Screenshots

![image](https://user-images.githubusercontent.com/444632/225768478-e77adcee-5856-42c6-b289-1475e9c4d867.png)




### How to test the changes in this Pull Request:


1. Enable the `block-editor-feature-enabled` feature flag.
2. Navigate to Product -> Add new to see the block editor.
3. Click the `...` more menu in the top right, ensure it appears correctly.
4. Run `localStorage.setItem( 'debug', 'wc-admin:*' );` in you console, and then click the "Share feedback" item in the more menu.
5. Leave feedback, and ensure the tracks event appears in the console.
6. Click "Use the classic editor" within the more menu.
7. Note that the modal appears. Close it and note that the classic editor appears.
8. Disable the previous feature flag, and then enable the `new-product-management-experience` feature flag.
9. Go to Product -> Add new to display the non-block new product editor. 
10. Repeat steps 5-7 for the more menu items. 



### Other information:




### FOR PR REVIEWER ONLY:
